app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/ColorCorrect.scala

```diff
@@ -153,25 +153,65 @@ object ColorCorrect {
     val maybeEqualize =
       if (params.equalize) Some(HistogramEqualization(_: MultibandTile, rgbHist)) else None
 
+    case class LayerClipping(redMin: Int, redMax: Int, greenMin: Int, greenMax:Int, blueMin: Int, blueMax: Int)
+    case class ClippingParams(band: Int, min: Option[Int], max: Option[Int], layerMin: Int, layerMax: Int)
+   
+    val layerRgbClipping = (for {
+      (r :: g :: b :: xs) <- Some(rgbHist.toList)
+      isCorrected   <- Some((r :: g :: b :: Nil).foldLeft(true) {(acc, hst) => (
+                          acc && (hst match {
+                            case x if 1 until 255 contains x.minValue().map(_.toInt).getOrElse(0) => true
+                            case x if 1 until 255 contains x.maxValue().map(_.toInt).getOrElse(0) => true
```

Comment:
I don't really have a strong preference, but I thought `0` was clearer since I didn't know whether `until` was inclusive. I can revise it to `255` though.